ibis/expr/datatypes/__init__.py

```diff
@@ -1074,18 +1074,24 @@ def parser():
         | spaceless_string("null").result(null)
         | spaceless_string("boolean").result(boolean)
         | spaceless_string("bool").result(boolean)
+        | spaceless_string("integer").result(int32)
         | spaceless_string("int8").result(int8)
+        | spaceless_string("smallint").result(int8)
         | spaceless_string("int16").result(int16)
         | spaceless_string("int32").result(int32)
         | spaceless_string("int64").result(int64)
+        | spaceless_string("bigint").result(int64)
+        | spaceless_string("uinteger").result(uint8)
         | spaceless_string("uint8").result(uint8)
         | spaceless_string("uint16").result(uint16)
         | spaceless_string("uint32").result(uint32)
         | spaceless_string("uint64").result(uint64)
+        | spaceless_string("ubigint").result(uint64)
         | spaceless_string("halffloat").result(halffloat)
         | spaceless_string("double").result(double)
         | spaceless_string("float16").result(float16)
         | spaceless_string("float32").result(float32)
+        | spaceless_string("real").result(float32)
```

Comment:
I think I have an alternative type parser laying around in an older version my branch on this, I'll push it up.

Reply:
ah, cool.  Ripping this out into a separate parser was on my todo list, but if it's already done, that's :man_cook: :kiss: